appserver/payara-appserver-modules/security-oauth2/src/main/java/fish/payara/security/oauth2/OAuth2AuthenticationMechanism.java

```diff
@@ -248,4 +264,29 @@ private AuthenticationStatus redirectForAuth(HttpMessageContext context) {
         return context.redirect(authTokenRequest.toString());
     }
 
+    
+    private String getConfiguredValue(String value, Config provider, String mpConfigKey){
+        String result = value;
+        Optional<String> configResult = provider.getOptionalValue(mpConfigKey, String.class);
+        if (configResult.isPresent()) {
+            result = configResult.get();
```

Comment:
If config result is present, just return it. Don't let that value go through the processors. 

(a concern was raised that this might introduce security loop holes, as the MP Config values can be set externally)